### PR TITLE
Set correct virtual model name in NEST-SONATA documentation

### PR DESCRIPTION
**Pull request by nicolossus**
NEST-SONATA uses the `spike_train_injector` as virtual model (used to be `spike_generator`). This PR changes the name of the virtual model to the correct one in places that were overlooked when the switch of models took place. The PR also improves the formatting by removing trailing whitespaces.

### CODE REVIEW

**Comment by nicolossus** [comment]
@jessica-mitchell The failing CI was due to a formatting error in master which was fixed by #2837. Could you retrigger the CI?

File: doc/htmldoc/nest_sonata/nest_sonata_guide.rst [hunk 1]
```diff
@@ -100,32 +100,33 @@ NEST assumes the following structure of the node HDF5 files:
...
 * ``point_process``
-* ``virtual`` 
+* ``virtual``
 
-Both ``point_neuron`` and ``point_process`` mean that the node is a neuron model (explicitly simulated; the two terms can be used interchangeably) whereas ``virtual`` 
-means that the node only provide inputs to the simulated system. ``virtual`` nodes are modeled as ``spike_generator``\s 
-(see :doc:`../models/spike_generator`\). NEST requires that only one ``model_type`` is present per node type CSV file. 
+Both ``point_neuron`` and ``point_process`` mean that the node is a neuron model (explicitly simulated; the two terms
+can be used interchangeably) whereas ``virtual`` means that the node only provide inputs to the simulated system.
+``virtual`` nodes are modeled as ``spike_train_injector``\s (see :doc:`../models/spike_train_injector`\). NEST requires
```
Comment (jessica-mitchell) [inline doc/htmldoc/nest_sonata/nest_sonata_guide.rst:113]:
```suggestion
``virtual`` nodes are modeled as ``spike_train_injector``\s (see :doc:`the model documentation for spike_train_injector <../models/spike_train_injector>`\). NEST requires
```

**Review by jessica-mitchell — CHANGES_REQUESTED** [review]
Looks good, just one minor suggestion

**Review by jessica-mitchell — APPROVED** [review]

**Comment by jessica-mitchell** [comment]
Merging with 1 reviewer as its doc related

**Pull request merged**